From 20a45e055d86c5655b74787b74ed78f13a6766da Mon Sep 17 00:00:00 2001
From: Bruce Guenter <bruce@untroubled.org>
Date: Wed, 28 Oct 2015 15:42:41 -0600
Subject: [PATCH] Allow writing stats to stdout with option -s -

---
 rbldnsd.c | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/rbldnsd.c b/rbldnsd.c
index abf1d01..f9c4356 100644
--- a/rbldnsd.c
+++ b/rbldnsd.c
@@ -682,13 +682,28 @@ struct dnsstats gstats;
 static struct dnsstats gptot;
 static time_t stats_time;
 
+static FILE *open_stats(void) {
+  return (statsfile[0] == '-' && statsfile[1] == 0)
+    ? stdout
+    : fopen(statsfile, "a");
+}
+
+static void close_stats(FILE *f) {
+  if (f) {
+    if (f == stdout)
+      fflush(f);
+    else
+      fclose(f);
+  }
+}
+
 static void dumpstats(void) {
   struct dnsstats tot;
   char name[DNS_MAXDOMAIN+1];
   FILE *f;
   struct zone *z;
 
-  f = fopen(statsfile, "a");
+  f = open_stats();
 
   if (f)
     fprintf(f, "%ld", (long)time(NULL));
@@ -720,7 +735,7 @@ static void dumpstats(void) {
       delta(q_ok), delta(q_nxd),
       delta(b_in), delta(b_out));
 #undef delta
-    fclose(f);
+    close_stats(f);
   }
   if (stats_relative)
     gptot = tot;
@@ -728,10 +743,10 @@ static void dumpstats(void) {
 }
 
 static void dumpstats_z(void) {
-  FILE *f = fopen(statsfile, "a");
+  FILE *f = open_stats();
   if (f) {
     fprintf(f, "%ld\n", (long)time(NULL));
-    fclose(f);
+    close_stats(f);
   }
 }
 
-- 
2.6.2

