diff --git a/gnome-terminal/zvtterm.c b/gnome-terminal/zvtterm.c
index 0d6f113..6939052 100644
--- a/gnome-terminal/zvtterm.c
+++ b/gnome-terminal/zvtterm.c
@@ -5423,7 +5423,7 @@ void mgt_draw_chars(ZvtTerm *term,
 	term->fnt_boxes = FALSE;
 	term->force_box_chars = save_force;
       }
-    else
+    else {
 #endif
 	for (i=0;i<len;i++) {
 	  expand[i] = VT_ASCII(line->data[i+col]) & 0xff;
@@ -5444,6 +5444,7 @@ void mgt_draw_chars(ZvtTerm *term,
 	  XmbDrawString(drawable_private->xdisplay, drawable_private->xwindow,
 			fontset, gc_private->xgc, offx + x + delx, offy + y + dely, expand, len);
     }
+    }
   }
  
 
