commit 961f6ad7990243d273795271cd1dd97b5a1c3832
Author: Bruce Guenter <bruce@untroubled.org>
Date:   Fri Jan 2 14:20:04 2015 -0600

    daemon: Support GOARCH 386 as well

diff --git a/daemon/daemon.go b/daemon/daemon.go
index 632b9ab..2fa628e 100644
--- a/daemon/daemon.go
+++ b/daemon/daemon.go
@@ -1114,8 +1114,8 @@ func (daemon *Daemon) ImageGetCached(imgID string, config *runconfig.Config) (*i
 
 func checkKernelAndArch() error {
 	// Check for unsupported architectures
-	if runtime.GOARCH != "amd64" {
-		return fmt.Errorf("The Docker runtime currently only supports amd64 (not %s). This will change in the future. Aborting.", runtime.GOARCH)
+	if runtime.GOARCH != "amd64" && runtime.GOARCH != "386" {
+		return fmt.Errorf("The Docker runtime currently only supports amd64 and 386 (not %s). This will change in the future. Aborting.", runtime.GOARCH)
 	}
 	// Check for unsupported kernel versions
 	// FIXME: it would be cleaner to not test for specific versions, but rather
