diff --git a/MAN b/MAN
index 1cc4fc4..aaf89e8 100644
--- a/MAN
+++ b/MAN
@@ -8,7 +8,12 @@ c:::644:/man8/:readproctitle.8:
 c:::644:/man8/:setlock.8:
 c:::644:/man8/:setuidgid.8:
 c:::644:/man8/:softlimit.8:
+c:::644:/man8/:supervise.8:
+c:::644:/man8/:svc.8:
+c:::644:/man8/:svok.8:
 c:::644:/man8/:svscan.8:
 c:::644:/man8/:svscanboot.8:
-c:::644:/man8/:tai64n.8
-c:::644:/man8/:tai64nlocal.8
+c:::644:/man8/:svstat.8:
+c:::644:/man8/:svup.8:
+c:::644:/man8/:tai64n.8:
+c:::644:/man8/:tai64nlocal.8:
